src/twisted/trial/reporter.py

```diff
@@ -107,9 +107,11 @@ def _getFailure(self, error):
         """
         Convert a C{sys.exc_info()}-style tuple to a L{Failure}, if necessary.
         """
-        if isinstance(error, tuple):
+        if isinstance(error, tuple) and len(error) == 3:
```

Comment:
Very minor comment.

Maybe add a comment to explain what `len(e) == 3` means,
or refactor the code to add a semantic to the condition.

```suggestion
        is_valid_tuple = isinstance(error, tuple) and len(error) == 3
        if is_valid_tuple:
```